api_app/api/routes/workspaces.py

```diff
@@ -234,3 +234,10 @@ async def delete_user_resource(workspace_id: str, service_id: str, resource_id:
     await send_uninstall_message(user_resource, user_resource_repo, previous_deletion_status, ResourceType.UserResource)
 
     return UserResourceIdInResponse(resourceId=user_resource.id)
+
+
+@user_resources_router.patch("/workspaces/{workspace_id}/workspace-services/{service_id}/user-resources/{resource_id}", response_model=UserResourceInResponse, name=strings.API_UPDATE_USER_RESOURCE)
+async def patch_user_resource(workspace_id: str, service_id: str, resource_id: str, user_resource_patch: UserResourcePatchEnabled, user_resource=Depends(get_user_resource_by_id_from_path), user_resource_repo=Depends(get_repository(UserResourceRepository)), user=Depends(get_current_user), workspace_service=Depends(get_deployed_workspace_service_by_id_from_path), workspace=Depends(get_workspace_by_workspace_id_from_path)) -> UserResourceInResponse:
+    validate_user_is_owner_or_researcher(user, workspace)
```

Comment:
For user resources, if the user is researcher, we need to check that they are actually the owner of the resource. You should not be able to patch other researchers resources -- but the workspace owner can patch all... See the delete functionality for the checks